src/codeManager.ts

```diff
@@ -37,15 +38,39 @@ export class CodeManager {
         }
 
         const iotHubConnectionString = Utility.getConnectionStringWithId(Constants.IotHubConnectionStringKey);
-        const content = fs.readFileSync(this.context.asAbsolutePath(path.join("resources", "code-template", Constants.CodeTemplates[language][type])), "utf8")
-            .replace(/{{deviceConnectionString}}/g, deviceItem.connectionString)
-            .replace(/{{iotHubConnectionString}}/g, iotHubConnectionString)
-            .replace(/{{deviceId}}/g, deviceItem.deviceId)
-            .replace(/{{iotHubHostName}}/g, Utility.getHostName(iotHubConnectionString))
-            .replace(/{{deviceSasToken}}/g, Utility.generateSasTokenForDevice(deviceItem.connectionString));
-        const textDocument = await vscode.workspace.openTextDocument({ content, language: Constants.LanguageIds[language] });
-        vscode.window.showTextDocument(textDocument);
+        const template = this.context.asAbsolutePath(path.join("resources", "code-template", Constants.CodeTemplates[language][type]));
+        const replacements = new Map(
+            [[/{{deviceConnectionString}}/g, deviceItem.connectionString],
+            [/{{iotHubConnectionString}}/g, iotHubConnectionString],
+            [/{{deviceId}}/g, deviceItem.deviceId],
+            [/{{iotHubHostName}}/g, Utility.getHostName(iotHubConnectionString)],
+            [/{{deviceSasToken}}/g, Utility.generateSasTokenForDevice(deviceItem.connectionString)],
+            ]);
+        if ((await fs.stat(template)).isFile()) {
+            let content = await fs.readFile(template, "utf8");
+            for (const [key, value] of replacements) {
+                content = content.replace(key, value);
+            }
+            const textDocument = await vscode.workspace.openTextDocument({ content, language: Constants.LanguageIds[language] });
+            vscode.window.showTextDocument(textDocument);
+        } else {
+            const folderUri: vscode.Uri[] = await vscode.window.showOpenDialog({
+                canSelectFiles: false,
+                canSelectFolders: true,
+            });
+            if (!folderUri) {
+                return;
+            }
+            const folder = folderUri[0].fsPath;
+            await fs.copy(template, folder);
+            await replace({
+                files: `${folder}/**/*`,
+                from: [...replacements.keys()],
+                to: [...replacements.values()],
+            });
+            await vscode.commands.executeCommand("vscode.openFolder", vscode.Uri.file(folder), true);

```

Comment:
This will open a new VS Code instance. In the future, we may consider providing choice to open in new window/current window/add to Workspace, just like Azure Function extension does.

Reply:
Let's discuss the exp with Xin?

Reply:
Synced with Xin, he is OK with this experience.